packages/react-forms/src/InputDate/index.js

```diff
@@ -21,141 +22,177 @@ function toYYYYMMDD(date) {
   return `${year}-${month}-${day}`;
 }
 
-type Props = {
+function isBoolean(bool) {
+  return bool === true || bool === false;
+}
+
+export type Props = {
   value: string,
-  onChange: Function,
+  onChange: string => void,
+  defaultIsOpen?: boolean,
   isOpen?: boolean,
-  onToggle?: Function,
+  onToggle?: boolean => void,
   min?: string,
   max?: string,
   disabled?: boolean,
+  defaultCalendarValue?: Date,
+  onCalendarChange?: Date => void,
+  calendarValue?: Date,
 };
 
-type State = {
-  current?: Date,
-  isOpen: boolean,
-};
-
-export default class InputDate extends React.Component<Props, State> {
-  component = React.createRef /*:: <HTMLElement> */();
-  refA = React.createRef /*:: <HTMLElement> */();
-  refB = React.createRef /*:: <HTMLElement> */();
-
-  state = {
-    current: undefined,
-    isOpen: false,
-  };
-
-  get isOpen(): boolean {
-    return Boolean(this.props.onToggle ? this.props.isOpen : this.state.isOpen);
-  }
-
-  setIsOpen = (isOpen: boolean) =>
-    this.props.onToggle
-      ? this.props.onToggle(isOpen)
-      : this.setState({ isOpen });
-
-  handleOpen = () => this.isOpen === false && this.setIsOpen(true);
-
-  handleClose = /* istanbul ignore next */ () =>
-    void (this.isOpen === true && this.setIsOpen(false));
-
-  handleCloseOnBlur = (evt: MouseEvent) => {
-    const { relatedTarget } = evt;
-    // istanbul ignore next (this is too DOM related)
-    if (
-      this.component.current && relatedTarget instanceof HTMLElement
-        ? !this.component.current.contains(relatedTarget)
-        : true
-    ) {
-      this.setIsOpen(false);
+const InputDate = ({
+  value,
+  onChange,
+  defaultIsOpen = false,
+  isOpen: isOpenProp,
+  onToggle,
+  min,
+  max,
+  defaultCalendarValue,
+  onCalendarChange,
+  calendarValue,
+  ...props
+}: Props) => {
+  const component = createRef();
+  const refA = createRef();
+  const refB = createRef();
+  const didMount = useRef();
+
+  const [isOpen, setOpen] = useControlledState(
+    isBoolean(isOpenProp) ? undefined : defaultIsOpen,
+    isOpenProp,
+    onToggle
+  );
+
+  const [current, setCurrent] = useControlledState(
+    calendarValue === undefined
```

Comment:
this doesn't look right, the user should either provide a `calendarValue` or a `defaultCalendarValue`, they are exclusive.

Reply:
Thought we could not make it mandatory. Should we?

Reply:
They can be non mandatory, but in that case you will define a default value to `defaultCalendarValue` in the object destructuring of the component.

Reply:
Okey right.